effekt/shared/src/main/scala/effekt/RecursiveDescent.scala

```diff
@@ -197,7 +210,7 @@ class RecursiveDescent(positions: Positions, tokens: Seq[Token], source: Source)
           semi()
           if returnPosition then Return(e)
           else ExprStmt(e, stmts())
-      }
+      }) labelled "statements"
```

Comment:
I think it's fine to merge it like this, but maybe we should also figure out a prefix version of `labelled`, this looks a bit weird for my taste.
(That's on me, I made the combinator like this :sweat_smile:)